source/common/network/tcp_listener_impl.cc

```diff
@@ -26,14 +26,38 @@ bool TcpListenerImpl::rejectCxOverGlobalLimit() const {
     return false;
   }
 
-  // If the connection limit is not set, don't limit the connections, but still track them.
-  // TODO(tonya11en): In integration tests, threadsafeSnapshot is necessary since the FakeUpstreams
-  // use a listener and do not run in a worker thread. In practice, this code path will always be
-  // run on a worker thread, but to prevent failed assertions in test environments, threadsafe
-  // snapshots must be used. This must be revisited.
-  const uint64_t global_cx_limit = runtime_.threadsafeSnapshot()->getInteger(
-      GlobalMaxCxRuntimeKey, std::numeric_limits<uint64_t>::max());
-  return AcceptedSocketImpl::acceptedSocketCount() >= global_cx_limit;
+  if (trackGlobalActiveCxLimitInOverloadManager()) {
+    // Check if deprecated runtime flag `overload.global_downstream_max_connections` is configured
+    // simultaneously with downstream connections monitor in overload manager.
+    if (runtime_.threadsafeSnapshot()->get(GlobalMaxCxRuntimeKey)) {
```

Comment:
One capability that the runtime key has the the existing mechanism doesn't is the ability to get values set during runtime. During deprecation we could see if that is being used by anyone.